test/boost/memtable_test.cc

```diff
@@ -1315,28 +1321,85 @@ SEASTAR_TEST_CASE(test_exceptions_in_flush_on_sstable_write) {
     };
     auto mep = std::make_unique<myext>();
     auto& me = *mep;
-    co_await exceptions_in_flush_helper(std::move(mep), me.should_fail, me.did_fail);
+    me.throw_func = std::move(throw_func);
+    co_await exceptions_in_flush_helper(std::move(mep), me.should_fail, me.did_fail, expect_isolate);
 }
 
-SEASTAR_TEST_CASE(test_exceptions_in_flush_on_sstable_open) {
+static auto test_case() {
+    return boost::unit_test::framework::current_test_case().p_name;
+}
+
+SEASTAR_TEST_CASE(test_exceptions_in_flush_on_sstable_write) {
+    co_await exceptions_in_flush_on_sstable_write_helper(
+        [] { throw std::system_error(EACCES, std::system_category()); }
+    );
+}
+
+SEASTAR_TEST_CASE(test_ext_permission_exceptions_in_flush_on_sstable_write) {
+    co_await exceptions_in_flush_on_sstable_write_helper(
+        [] { throw db::extension_storage_permission_error(test_case()); }
```

Comment:
could just put
```c++
[] { throw db::extension_storage_permission_error(get_name()); }
```
without introducing the helper, because it's already implemented, 
see https://github.com/scylladb/seastar/blob/3193c52e0c2ffc0db8c08904219bfcf822d5a60b/include/seastar/testing/seastar_test.hh#L55 .

Reply:
Neat. Done.